chardet/charsetprober.py

```diff
@@ -79,8 +78,8 @@ def filter_international_words(buf):
 
         This filter applies to all scripts which do not use English characters.
         """
-        filtered = BytesIO()
 
+        out = b''
```

Comment:
Your suspicion is right for Python 3 but wrong for Python 2.7. 
It is not good idea to create one project for various python's versions. There are many problems with compatibility and speed optimization.